test/testdata/rewriter/chained_sig.rb

```diff
@@ -0,0 +1,208 @@
+  sig.void
+    # ^^^^ error: Cannot use `void` outside of a sig block
+    # ^^^^ error: Method `void` does not exist on `NilClass`
+  def baz; end
+
+  sig.checked(:never)
+    # ^^^^^^^ error: Cannot use `checked` outside of a sig block
+    # ^^^^^^^ error: Method `checked` does not exist on `NilClass`
+  def blah; end
+
+  sig.on_failure(:soft, notify: "me")
+    # ^^^^^^^^^^ error: Cannot use `on_failure` outside of a sig block
+    # ^^^^^^^^^^ error: Method `on_failure` does not exist on `NilClass`
+  def bip; end
+
+  sig.type_parameters(:T)
+    # ^^^^^^^^^^^^^^^ error: Cannot use `type_parameters` outside of a sig block
+    # ^^^^^^^^^^^^^^^ error: Method `type_parameters` does not exist on `NilClass`
+  def bop; end
+end
+
+class DuplicateBlock
+  extend T::Sig
+  include Interface
+
+  sig.override { void }.final { void }
+                      # ^^^^^ error: Cannot add more signature statements after the declaration block
+  def foo; end
+end
+
+class WithoutRuntime
+  extend T::Sig
+  include Interface
+
+  T::Sig::WithoutRuntime.sig.final { void }
+  def baz; end
+
+  T::Sig::WithoutRuntime.sig.params(a: Integer)
+                           # ^^^^^^ error: Cannot use `params` outside of a sig block
+                           # ^^^^^^ error: Method `params` does not exist on `NilClass`
+  def bar(a); end
+
+  T::Sig::WithoutRuntime.sig.override { void }.final { void }
+                                             # ^^^^^ error: Cannot add more signature statements after the declaration block
+  def foo; end
+end
+
+class MissingBlocks
+  extend T::Sig
+
+  sig
+# ^^^ error: Signature declarations expect a block
+  def foo; end
+
+  sig.final
+# ^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+  def bar; end
+
+  sig.abstract
+# ^^^^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^^^^ error: Method `abstract` does not exist on `NilClass`
+  def baz; end
+
+  sig.override
+# ^^^^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^^^^ error: Method `override` does not exist on `NilClass`
+  def qux; end
+
+  sig.overridable
+# ^^^^^^^^^^^^^^^ error: Signature declarations expect a block
+    # ^^^^^^^^^^^ error: Method `overridable` does not exist on `NilClass`
+  def quux; end
+end
+
+class InstructionSequenceInSig
+  extend T::Sig
+
+  sig.final { Kernel.puts ""; void }
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Malformed signature: cannot have multiple statements inside a signature block
+    # ^^^^^ error: Method `final` does not exist on `NilClass`
+                            # ^^^^ error: Method `void` does not exist on `T.class_of(InstructionSequenceInSig)`
+  def foo; end
+end
+
+class DuplicateInvocations
+  extend T::Sig
+  extend T::Helpers
+  include Interface
+
+  abstract!
+
+  sig.overridable { overridable.void }
+                  # ^^^^^^^^^^^ error: Duplicate invocation of `overridable` in signature declaration
```

Comment:
`resolver/type_syntax.cc` is already checking for duplicate invocations of some (but not all) methods.

WDYT about making `type_syntax.cc`'s checking more complete (as a separate PR), and then we don't have to have the duplicate checks in the `ChainedSig` rewriter?  Ideally the locations to get good error messages will just happen naturally with chained sigs too.